enhancements/windows-containers/health-management.md

```diff
@@ -0,0 +1,258 @@
+  existing ability to properly configure Windows VMs to join the cluster as Windows Nodes.
+
+
+### Non-Goals
+
+* Diagnosing or remediating generic Windows issues, unrelated to changes made by WMCO.
+* Cluster-level resource management, such as deleting or re-creating WMCO-configured Machines that enter an unrecoverable state (this is a responsibility of [MachineHealthChecks](https://docs.openshift.com/container-platform/latest/machine_management/deploying-machine-health-checks.html)).
+
+
+## Proposal
+
+Currently, Windows Node configuration process is split between two components, WMCO and [WMCB](https://github.com/openshift/windows-machine-config-operator/).
+WMCB is a one shot program, which is copied to and subsequently ran on Windows VMs, and in order to configure and
+start the Kubelet Windows Service.
+
+In order to accomplish the goals listed in this enhancement, I am proposing that WMCB be converted into a daemon,
+Windows Machine Config Daemon (WMCD). WMCD will have the following responsibilities:
+* WMCD maintains current responsibilities of bootstrapping kubelet through the `bootstrap` command.
+* Using the configuration provided by a ConfigMap created by WMCO, WMCD maintains the state of Windows Services on the VM.
+* WMCD cleans up all changes it has made to a VM when ran with the `deconfigure` command.
+
+The Windows Node creation workflow for the end user should not be changed.
+
+### Services ConfigMap
+
+WMCO will create and maintain an immutable ConfigMap with the naming scheme:
+`services-<MajorVersion>-<MinorVersion>-<PatchVersion>-<Commit>`. For example, if WMCO was built from commit a7b5
+of WMCO version 5.0.0, the created ConfigMap would be named `config-5-0-0-a7b5`.
+
+The purpose of this ConfigMap is to provide WMCD with the specifications required to configure a Node. The ConfigMap
+will be watched via the existing ConfigMap controller. If it is created with incorrect values, WMCO will delete it.
+If it is deleted, WMCO will re-create it. If it is not present when WMCO is kicking off the configuration of a VM,
+it will be created.
+
+The services ConfigMap contains two keys: `services`, and `files`. The values for both keys are JSON objects.
+The services key contains all data required to configure the required Windows services on any VM that is to be
+added to the cluster as a Node. The proposed schema is as follows:
+```json
+[
+  {
+    "name": "name of the Windows service",
+    "command": "the command that will be ran",
+    "nodeVariablesToSubstitute": {
+      "name": "string within command field that will be substituted",
+      "jsonPathNodeObject": "value that will be substituted, taken from the provided jsonPath of the Node object"
+    },
+    "powershellVariablesToSubstitute": {
+      "name": "string within command field that will be substituted",
+      "path": "value that will be substituted, taken from the result of the powershell script at this path"
```

Comment:
The explanation is confusing and needs to be reworded. `path` is just the location of the PS script.